packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -435,24 +453,30 @@ export default function(
     adoptClassInstance(workInProgress, instance);
 
     if (__DEV__) {
-      if (typeof ctor.getDerivedStateFromProps === 'function') {
-        if (state === null) {
-          const componentName = getComponentName(workInProgress) || 'Component';
-          if (!didWarnAboutUninitializedState[componentName]) {
-            warning(
-              false,
-              '%s: Did not properly initialize state during construction. ' +
-                'Expected state to be an object, but it was %s.',
-              componentName,
-              instance.state === null ? 'null' : 'undefined',
-            );
-            didWarnAboutUninitializedState[componentName] = true;
-          }
+      if (
+        typeof ctor.getDerivedStateFromProps === 'function' &&
+        state === null
+      ) {
+        const componentName = getComponentName(workInProgress) || 'Component';
+        if (!didWarnAboutUninitializedState[componentName]) {
+          warning(
+            false,
+            '%s: Did not properly initialize state during construction. ' +
+              'Expected state to be an object, but it was %s.',
+            componentName,
+            instance.state === null ? 'null' : 'undefined',
+          );
+          didWarnAboutUninitializedState[componentName] = true;
         }
+      }
 
-        // If getDerivedStateFromProps() is defined, "unsafe" lifecycles won't be called.
-        // Warn about these lifecycles if they are present.
-        // Don't warn about react-lifecycles-compat polyfilled methods though.
+      // If new component APIs are defined, "unsafe" lifecycles won't be called.
+      // Warn about these lifecycles if they are present.
```

Comment:
We're sure we won't forget to fire this warning at FB? Since it would be pretty essential to know this happened.

Reply:
I think you're just saying that we should be sure to update the warning whitelist?

Reply:
Maybe, I'm just not sure if this warning is also coalesced, and if it is, whether it will be in the whitelist or not.

Reply:
This warning isn't coalesced. We log once per component-type if there is a combo of new and unsafe lifecycles. We'll need to add this pattern to the `createWarning` list though for sure. It isn't there currently.